### DISCUSSION TITLE
Nothing is looping with useTypewriter hook

### DISCUSSION THREAD

**Discussion by Yazan-Egbaria** [community/community#119783] [Programming Help] [answered]
### Body

Hey,

So this is the component js
Why is nothing looping? even nothing is rendering, and when i log text it says undefined? though the react simple typewriter is installed

```
import "./Header.css";
import Headercontent from "../../components/Headercontent/Headercontent";
import { useTypewriter, Cursor } from "react-simple-typewriter";

const Header = () => {
  const { text } = useTypewriter({
    words: ["Developer", "Designer", "Yazan"],
    loop: true,
    typeSpeed: 80,
    deleteSpeed: 80,
  });

  return (
    <div className="Header">
      <div className="Wrapper">
        <Headercontent
          className="text-container"
          span="Heya!"
          header={
            <>
              I am <span className="span">{text}</span>
              <span style={{ color: "#C9DA2D" }}>
                <Cursor />
              </span>
            </>
          }
        />
      </div>
    </div>
  );
};

export default Header;
```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ThisaraJayas** [comment]
Try this:


import React from "react";
import "./Header.css";
import Headercontent from "../../components/Headercontent/Headercontent";
import { useTypewriter, Cursor } from "react-simple-typewriter";

const Header = () => {
  const { text } = useTypewriter({
    words: ["Developer", "Designer", "Yazan"],
    loop: true,
    typeSpeed: 80,
    deleteSpeed: 80,
  });

  return (
    <div className="Header">
      <div className="Wrapper">
        <Headercontent
          className="text-container"
          span="Heya!"
          header={
            <>
              I am <span className="span">{text}</span>
              <span style={{ color: "#C9DA2D" }}>
                <Cursor />
              </span>
            </>
          }
        />
      </div>
    </div>
  );
};

export default Header;

**Comment by codewithshinde** [comment, marked as answer]
Ohh !! Simple all you need to do is change {text} to [text]
you can also check the same here :  [Example in Sandbox](https://codesandbox.io/p/sandbox/react-simple-typewriter-example-dtcy8n?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clv8sktyr0006356ies6bus6p%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clv8sktyr0002356iebsfwf7z%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clv8sktyr0003356i6zmghlus%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clv8sktyr0005356iy6td1413%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clv8sktyr0002356iebsfwf7z%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clv8sktyr0001356izrncbja1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%257D%255D%252C%2522id%2522%253A%2522clv8sktyr0002356iebsfwf7z%2522%252C%2522activeTabId%2522%253A%2522clv8sktyr0001356izrncbja1%2522%257D%252C%2522clv8sktyr0005356iy6td1413%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clv8sktyr0004356i0j9pf50p%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clv8sktyr0005356iy6td1413%2522%252C%2522activeTabId%2522%253A%2522clv8sktyr0004356i0j9pf50p%2522%257D%252C%2522clv8sktyr0003356i6zmghlus%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clv8sktyr0003356i6zmghlus%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)
```javascript
import "./Header.css";
import Headercontent from "../../components/Headercontent/Headercontent";
import { useTypewriter, Cursor } from "react-simple-typewriter";

const Header = () => {
  const [ text ] = useTypewriter({
    words: ["Developer", "Designer", "Yazan"],
    loop: true,
    typeSpeed: 80,
    deleteSpeed: 80,
  });
